src/zope/testrunner/find.py

```diff
@@ -418,11 +442,19 @@ def tests_from_suite(suite, options, dlevel=1,
     if isinstance(suite, unittest.TestSuite):
         for possible_suite in suite:
             for r in tests_from_suite(possible_suite, options, level, layer,
-                                      accept=accept):
+                                      accept=accept,
+                                      seen_test_ids=seen_test_ids,
+                                      duplicated_test_ids=duplicated_test_ids):
                 yield r
     elif isinstance(suite, StartUpFailure):
         yield (suite, None)
     else:
+        if options.require_unique_ids:
+            suite_id = suite.id()
+            if suite_id in seen_test_ids:
+                duplicated_test_ids.add(suite_id)
```

Comment:
(Augh, GitHub cuts off the one line I hoped would be visible above my comment.)

TL;DR: I think you should change this to `suite_id = str(suite)`, and it's all good!